updating/understanding-openshift-upgrade-duration.adoc

```diff
@@ -0,0 +1,91 @@
+
+== Cluster upgrade duration
+
+Historical upgrade duration of similar clusters provides the best estimate for the future upgrades. However, if the historical data is not available, you can refer the below table to estimate the time duration of cluster update.
+
+The following table summarizes the approximate time duration of cluster update components:
+
+[options="header"]
+|===
+|Component Name |Time Duration (in minutes)
+
+|Cluster Version Operator (CVO)
+|45-120
+
+|Machine Config Operator (MCO)
+|5 per node
+|===
+
+=== Cluster Version Operator target upgrade payload deployment
+The Cluster Version Operator (CVO) unpacks the target upgrade release image and applies to the cluster. This process might take somewhere between 45 to 120 minutes.
+
+=== Machine Config Operator
+The Machine Config Operator (MCO) applies a new machine configuration to each control plane and worker node. During this process, the MCO performs the following tasks on each node of the cluster:
+
+* Node draining
+* OS update staging
+* Node rebooting
+
+The control plane node upgrade duration is more predictable and oftentimes shorter than worker nodes since it does not include large workloads. You can upgrade the worker nodes in parallel by setting the `maxUnavailable` field to greater than 1 in the machine configuration pool. The MCO cordons the number of nodes specified in `maxUnavailable` and marks them unavailable for upgrade.
+
+[IMPORTANT]
+====
+With {product-title} 4 upgrades, an operating system lifecycle management and maintenance is made as a part of the cluster upgrade.  This only applies to {op-system-first} nodes.
+====
+
+The time to complete this process depends on many factors including the node and infrastructure configuration. This process might take 5 minutes to complete per node.
+
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../architecture/architecture-installation.html#update-service-overview_architecture-installation[About the OpenShift Update Service]
+
+
+== Factors affecting the upgrade duration
+In addition to Cluster Version Operator (CVO) and Machine Config Operator (MCO), the following factors also impact the upgrade duration:
+
+* Pod Disruption Budget (PDB)
+* Cluster node health
+* Number of nodes in the cluster
+* Red Hat Enterprise Linux (RHEL) nodes
+
+=== Pod disruption budget
+You must consider pod disruption budgets and pod anti-affinity when you configure `maxUnavailable` field in the machine configuration pool.
+
+For example, if there is a cluster with three worker nodes, you must configure the disruption budget such that it has minimum two replicas be always running. When you configure pod anti-affinity, you must specify that two replicas should never run on the same node. To avoid pod scheduling issues, you must ensure the availability of minimum two worker nodes.
```

Comment:
> For example, if there is a cluster with three worker nodes, you must configure the disruption budget such that it has minimum two replicas be always running.

Folks can configure PDBs however they like, or skip PDBs entirely if they don't need to be highly available, so I'm not a fan of "you must configure" here.  If you have three possible host nodes, and strong anti-affinity, a PDB that requires three replicas might wedge the pool roll.  But that's setting a cap on PDB-required replicas, not a floor.  Something like:

> Increasing `maxUnavailable` on the machine configuration pool may help the pool update more quickly.  But if `maxUnavailable` is set too high, and several nodes are cordoned simultaneously, and there may be PDB-guarded workloads that fail to drain because replicas are unable to find a schedulable node to run on.  If you increase `maxUnavailable` for the machine configuration pool, ensure that you will still have sufficient schedulable nodes to allow PDB-guarded workloads to drain.

?

Reply:
@LalatenduMohanty Could you please look into this ? Is it possible to make this example more easy to readable?